packages/extension-chakra-storefront/src/setup-app.tsx

```diff
@@ -117,6 +117,11 @@ class ChakraStorefront extends ApplicationExtension<Config> {
             ignoredRoutes: ['/callback']
         })
     }
+
+    async getComponentMap() {
+        const modules = await import('./pages')
+        return this.generateComponentMapFromModules(modules)
+    }
```

Comment:
extension classes will need to implement the `getComponentMap` function as putting this logic in `ApplicationExtension.getComponentMap` gives errors when attempting to import from `./pages` since this folder only exists in extensions

Reply:
Can you remind me again why this is async?

Reply:
Good point this does not have to be async!